File: go/isogram/isogram.go
package isogram

import "unicode"

// IsIsogram returns true if a word is an isogram
func IsIsogram(word string) bool {
	set := make(map[rune]bool)
	for _, v := range word {
		if v == '-' || v == ' ' {
			continue
		}
		v = unicode.ToLower(v)
		if set[v] {
			return false
		}
		set[v] = true
	}
	return true
}

File: c/isogram/src/isogram.c
#include "isogram.h"
#include <ctype.h>
#include <string.h>

bool is_isogram(const char phrase[])
{
    if (phrase == NULL)
    {
        return false;
    }
    bool letters[26] = { false };
    for (size_t i = 0, n = strlen(phrase); i < n; i++)
    {
        char c = tolower((unsigned char) phrase[i]);
        if (!isalpha((unsigned char) c))
        {
            continue;
        }
        if (letters[c - 'a'])
        {
            return false;
        }
        letters[c - 'a'] = true;
    }
    return true;
}
